libraries/AP_Mount/AP_Mount_CADDX.h

```diff
@@ -0,0 +1,55 @@
+/*
+  CADDX mount using serial protocol backend class
+ */
+#pragma once
+
+#include "AP_Mount_config.h"
+
+#if HAL_MOUNT_CADDX_ENABLED
+
+#include "AP_Mount_Backend_Serial.h"
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_Common/AP_Common.h>
+
+class AP_Mount_CADDX : public AP_Mount_Backend_Serial
+{
+
+public:
+    // Constructor
+    using AP_Mount_Backend_Serial::AP_Mount_Backend_Serial;
+
+    // update mount position - should be called periodically
+    void update() override;
+
+    // has_pan_control - returns true if this mount can control its pan (required for multicopters)
+    bool has_pan_control() const override { return yaw_range_valid(); };
+
+    // has_roll_control - returns true if this mount can control its roll
+    bool has_roll_control() const override { return roll_range_valid(); };
+
+    // has_pitch_control - returns true if this mount can control its tilt
+    bool has_pitch_control() const override { return pitch_range_valid(); };
+
+protected:
+
+    // get attitude as a quaternion.  returns true on success
+    bool get_attitude_quaternion(Quaternion& att_quat) override;
+
+private:
+
+    // lock mode enum
+    enum class LockMode {
+        FOLLOW = 0,
```

Comment:
This doesn't make a lot of sense!
```suggestion
```

Reply:
Hi @peterbarker, thanks for the review.  I've removed this line because we don't use it but I can imagine that we could use it as a default value so not sure why it doesn't makes sense?

Reply:
Generally we don't use the same enumeration which is giving names to bit-offsets for *combinations* of those bits (AKA a "mask").

It would be nice to have some type-safety around this sort of thing, but I've never managed to propose a PR to add that....